app/views/recipes/_ingredient_fields.html.erb

```diff
@@ -1,29 +1,27 @@
-<% id = ingredient.id || Time.now.to_i %>
-
-<div id="ingredient_<%= id %>">
-  <%= fields_for "recipe[ingredients_attributes][#{id}]", ingredient do |form| %>
-    <%= form.hidden_field :id %>
-    <%= form.hidden_field :_destroy, value: local_assigns[:destroy] || false %>
-
-    <% unless local_assigns[:destroy] %>
-      <div
-        class="grid grid-cols-3 gap-6"
-        data-controller="streams"
-        data-streams-url-value="<%= nested_ingredient_path(id, format: :turbo_stream) %>">
-        <div class="col-span-3">
-          <div class="flex rounded-md shadow-sm relative">
-            <%= form.text_field :name, class: "block w-full flex-1 rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm", placeholder: "Eggs" %>
-            <% if local_assigns[:allow_remove] || ingredient.persisted? %>
-              <button data-action="streams#destroyRequest" type="button" class="bg-red-100 absolute w-10 border border-gray-300 rounded-r-md right-0 top-0 bottom-0 flex items-center justify-center">
-                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-4 h-4">
-                  <path stroke-linecap="round" stroke-linejoin="round" d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0" />
-                </svg>
-              </button>
-            <% end %>
-          </div>
-        </div>
-      </div>
-    <% end %>
-  <% end %>
-</div>
+<div id="<%= dom_id(form.object) %>" data-controller="visibility"
+     class="grid grid-cols-3 gap-6">
+  <div class="col-span-3">
+    <div class="flex rounded-md shadow-sm relative">
+      <%= form.text_field :name, class: "block w-full flex-1 rounded-md border-gray-300 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm", placeholder: "Eggs" %>
 
+      <% if form.object.persisted? %>
```

Comment:
The ingredients field will only need to know wether the form's object (the ingredient) has been persisted or not.

If it is, it renders a link with method `:delete` to use a turbo stream to remove it from the dom

If it's not, it means that it doesn't really need to go all the way to the server to remove it from the dom, so the same button uses the stimulus visibility controller to just remove it from the dom